src/text.rs

```diff
@@ -28,6 +30,25 @@ impl Text {
 
 impl AsRef<CStr> for Text {
     fn as_ref(&self) -> &CStr {
```

Comment:
> I'm not entirely sure that there should be an AsRef implementation for it in the first place, however, as creating a CStr (currently) requires walking the entire string every time. While this isn't the most expensive thing in the world, it's not 0-cost and (probably) a bad idea to conflate, especially since it may be done repeatedly when using them as references instead of once.

> In my opinion it should probably be changed to an Into<CStr> or similar.

You're right

https://doc.rust-lang.org/stable/src/core/ffi/c_str.rs.html#260

I'll probably get that changed, but it will be a breaking change.

Reply:
Changed in this commit. 93b08ccdd5c782fd058ed05d8db5ede4721d7a50

I didn't go for `Into`, as that is for owned->owned

https://rust-lang.github.io/api-guidelines/naming.html